server/controllers/CommentController.js

```diff
@@ -0,0 +1,145 @@
+import db from '../models';
+import response from '../helpers/response';
+import TokenManager from '../helpers/TokenManager';
+
+
+const { Article, Comment } = db;
+
+/**
+ * @class UserController
+ */
+class CommentController {
+  /**
+   *
+   * @description Allow user to comment on an article
+   * @static
+   * @param {*} req Express Request object
+   * @param {*} res Express Response object
+   * @returns {object} Json response
+   * @memberof CommentController
+   */
+  static async addComment(req, res) {
+    try {
+      const token = req.headers.authorization.split(' ')[1];
+      const { userId } = TokenManager.verify(token);
+      const { content } = req.body;
+      const { articleId } = req.params;
+      if (!content) {
+        return response(res, 400, 'failure', 'Enter a comment', null, null);
+      }
+      await Article.findAndCountAll({
+        where: {
+          id: articleId
+        }
+      });
+      const commentCreated = await Comment.create({
+        content,
+        userId,
+        articleId
+      });
+      if (commentCreated) {
+        return response(res, 201, 'success', 'Comment created', null, null);
+      }
+    } catch (error) {
+      return response(res, 404, 'failure', 'Article with the id not found', null, null);
+    }
+  }
+
+  /**
+   *
+   * @description Retrive all comment for an article
+   * @static
+   * @param {*} req Express Request object
+   * @param {*} res Express Response object
+   * @returns {object} Json response
+   * @memberof CommentController
+   */
+  static async getComments(req, res) {
+    try {
+      const { articleId } = req.params;
+      const commentsQuery = await Comment.findAll({
+        where: {
+          articleId
+        }
+      });
+      const comments = [];
+      commentsQuery.map(comment => comments.push(comment.dataValues));
+      return response(res, 200, 'success', 'Comment found', null, comments);
+    } catch (error) {
+      return response(res, 404, 'failure', 'Comment not found for article id', null, null);
```

Comment:
I think what we should have in our catch block should be an actual error (something that is an instance of the Error object). A poor 404 response shouldn't be subjected to such treatment (considered an error), conventionally speaking.

Reply:
Thanks have fix that with a 500 server error.